### PR TITLE
Feature/custom strings

### PR DESCRIPTION
**Pull request by b-scharbau**
Added StringProvider as optional parameters to the DayView() widget to allow for customization of the time and date display.

Please let me know if you need any additional fixes there

### CODE REVIEW

File: lib/src/components/day_view_components.dart [hunk 1]
```diff
@@ -140,13 +143,17 @@ class DefaultTimeLineMark extends StatelessWidget {
   /// Defines time to display
   final DateTime date;
 
+  /// StringProvider for time string
+  final StringProvider timeStringBuilder;
```
Comment (PRBaraiya) [inline lib/src/components/day_view_components.dart:147]: Make this parameter nullable.

File: lib/src/components/day_view_components.dart [hunk 2]
```diff
@@ -140,13 +143,17 @@ class DefaultTimeLineMark extends StatelessWidget {
...
+
   /// Text style for time string.
   final TextStyle? markingStyle;
 
   /// Time marker for timeline used in week and day view.
   const DefaultTimeLineMark({
     Key? key,
     required this.date,
+    required this.timeStringBuilder,
```
Comment (PRBaraiya) [inline lib/src/components/day_view_components.dart:156]: Make this optional.

File: lib/src/components/day_view_components.dart [hunk 3]
```diff
@@ -157,7 +164,7 @@ class DefaultTimeLineMark extends StatelessWidget {
       child: Padding(
         padding: const EdgeInsets.only(right: 7.0),
         child: Text(
-          "${((date.hour - 1) % 12) + 1} ${date.hour ~/ 12 == 0 ? "am" : "pm"}",
+          timeStringBuilder(date),
```
Comment (PRBaraiya) [inline lib/src/components/day_view_components.dart:167]: Check for null condition here. and use previous format if timeStringBuilder is null.

File: lib/src/day_view/day_view.dart [hunk 4]
```diff
@@ -343,7 +353,12 @@ class DayViewState<T> extends State<DayView<T>> {
   /// Default timeline builder this builder will be used if
   /// [widget.eventTileBuilder] is null
   ///
-  Widget _defaultTimeLineBuilder(date) => DefaultTimeLineMark(date: date);
+  Widget _defaultTimeLineBuilder(date) => DefaultTimeLineMark(
+      date: date,
+      timeStringBuilder: widget.timeStringBuilder ??
```
Comment (PRBaraiya) [inline lib/src/day_view/day_view.dart:358]: remove this null check from here.

**Review by PRBaraiya — CHANGES_REQUESTED** [review]

**Comment by PRBaraiya** [comment]
@b-scharbau  Thanks for raising this PR. I have requested some changes in this PR please review them.

**Comment by b-scharbau** [comment]
Sorry for late response, I was actually busy with other projects and didn't find the time to look after the requested changes.
Updated the code accordingly, please review

**Review by PRBaraiya — APPROVED** [review]

**Pull request merged**